kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -246,6 +241,7 @@ class BatchJobSubmission(
       throw new RuntimeException(s"$batchType batch[$batchId] job failed:" +
         applicationStatus.get.mkString(","))
     } else {
+      updateBatchMetadata()
```

Comment:
if the batch is recovered and before its state is PENDING.

We need update the state to RUNNING if get applicationId from resource manager before recovery.